django_components/component.py

```diff
@@ -24,9 +24,8 @@ class TokenType:
 
 
 class Component(with_metaclass(MediaDefiningClass)):
-    __cached_slot_nodes = None
 
-    def __init__(self, component_name):
+    def __init__(self, component_name='unnamed component'):
```

Comment:
Why would you want to create a component without a name?

Reply:
The reason I made this change was to let the dependency-rendering tags create components to access their media without needing to pass names for them.  Since the name is only used for informational purposes, it seemed easier to create a default rather than making changes there.

Understood on the need for more focused commits--I'm self-taught and self-employed, so being a good collaborator is definitely an area I can grow.  I appreciate your patience and the feedback.

Reply:
Hmm... I'm still not sure I follow. Doesn't the component registering happen before dependency rendering? If it does, shouldn't the dependency rendering code be able to get the names and the classes from the registry? Kinda funny that I'm a bit lost in the library I once wrote :) 

I'm only giving the feedback about focused commits because I've got that feedback myself, and found it very useful. I usually am all over the place when coding, fixing things as they come up, and therefore ending up with loads of changed code when it's time to commit. Two strategies I've found to deal with this:

- **My strategy**: I've learnt to add a step after my coding frenzies where I pick out changes line by line into commits before pushing a PR. I'm happy with the Tower App for this, but the Github App is good too. Command line is horrible for this.
- **Better strategy**: The best programmers I've worked with are using a "parking list". It's an open text file where they note down things they should "fix later". So when they encounter changes not related to the current thing they are working on, they add it there and stay focused on their original task. That specific task is what goes into the commit message (or ideally, the reason that task needs fixing). Seems simple, but requires immense self control that I hope to reach sometime :) 

Anyway, all these things are nice extras in the scheme of things. I'm super thankful that you are taking time from your busy day to help me make django-components great. I have a secret dream to merge it into Django one day :)

Reply:
Thanks, this is really helpful advice.  I'd love to see this library get folded into the mothership... it's really strange that there are so many different ways to include and inherit and modify templates in Django, but there's this huge gap for slots.

Anyways, I just pushed some changes that make the name argument mandatory.  I'd forgotten this nuance before I got back into the code, but the reason it's a little more complicated than just using the registry name is that component classes can be aliased in the registry.  I took the easy way out of just using the class `__name__` rather than trying to ensure that the names are unique--let me know if you want me to take a different approach.